ksql-engine/src/main/java/io/confluent/ksql/function/udtf/array/ExplodeTableFunction.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udtf.array;
+
+import io.confluent.ksql.function.BaseTableFunction;
+import io.confluent.ksql.name.FunctionName;
+import java.util.Collections;
+import java.util.List;
+import org.apache.kafka.connect.data.Schema;
+
+public class ExplodeTableFunction<T> extends BaseTableFunction<T, T> {
+
+  public ExplodeTableFunction(
+      final FunctionName functionName,
+      final Schema outputType,
+      final List<Schema> arguments,
+      final String description) {
+    super(functionName, outputType, arguments, description);
+  }
+
+  @Override
+  public List<T> flatMap(final List<T> input) {
+    return input == null ? Collections.emptyList() : input;
```

Comment:
nit: prefer `ImmutableList.of()` over `Collectections.emptyList()` and `ImmutableMap` is smart when it comes to making copies of other `ImmutableList`, but not `emptyList`s.

Reply:
This is being passed to KStreams so there seems no advantage to doing this. https://stackoverflow.com/questions/43291202/collections-emptylist-vs-guavas-immutablelist-of